packages/circuits/scripts/buildSnarks.mjs

```diff
@@ -24,35 +26,47 @@ for (const name of circuits) {
     if (!circuitContents[name])
         throw new Error(`Unknown circuit name: "${name}"`)
 
-    await fs.promises.writeFile(
-        path.join(outDir, `${name}_main.circom`),
-        circuitContents[name]
-    )
-
-    const inputFile = path.join(outDir, `${name}_main.circom`)
+    const inputFileBuild = path.join(buildDir, `${name}_main.circom`)
+    const inputFileOut = path.join(outDir, `${name}_main.circom`)
     const circuitOut = path.join(outDir, `${name}_main.r1cs`)
-    const wasmOut = path.join(outDir, `${name}_main_js/${name}_main.wasm`)
-    const wasmOutDir = path.join(outDir, `${name}_main_js`)
+    const circuitBuild = path.join(buildDir, `${name}_main.r1cs`)
+    const wasmOut = path.join(buildDir, `${name}_main_js/${name}_main.wasm`)
+    const wasmOutDir = path.join(buildDir, `${name}_main_js`)
     const wasmOutFinal = path.join(outDir, `${name}.wasm`)
+    const ptauBuild = path.join(buildDir, ptauName)
+    const zkeyBuild = path.join(buildDir, `${name}.zkey`)
+    const vkOutBuild = path.join(buildDir, `${name}.vkey.json`)
     const ptau = path.join(outDir, ptauName)
     const zkey = path.join(outDir, `${name}.zkey`)
     const vkOut = path.join(outDir, `${name}.vkey.json`)
 
-    // Check if the circuitOut file exists
     const circuitOutFileExists = await fs.promises
         .stat(circuitOut)
         .catch(() => false)
-    if (circuitOutFileExists) {
+    const zkeyOutFileExists = await fs.promises.stat(zkey).catch(() => false)
+    const vkeyOutFileExists = await fs.promises.stat(vkOut).catch(() => false)
+    const wasmOutFileExists = await fs.promises
+        .stat(wasmOutFinal)
+        .catch(() => false)
+
+    await fs.promises.writeFile(
+        path.join(buildDir, `${name}_main.circom`),
```

Comment:
This is confusing, there's no need to ever write this file to the build directory. You can just write it to `inputFileOut` instead of writing to build then copying

Reply:
Writing `circuitContents` directly to `outDir`